pkg/controller/endpointslice/reconciler.go

```diff
@@ -251,6 +258,33 @@ func (r *reconciler) reconcileByAddressType(service *corev1.Service, pods []*cor
 	return r.finalize(service, slicesToCreate, slicesToUpdate, slicesToDelete, triggerTime)
 }
 
+// placeholderSliceEqual returns if two placeholder slices are equal.
+// It ignores many of the fields in ObjectMeta that are not expected to match.
+func placeholderSliceEqual(x, y *discovery.EndpointSlice) bool {
+	if x == nil || y == nil {
+		return x == y
+	}
+
+	// Make a shallow copy (cheap) and wipe all metadata fields except the
+	// ones we actually care about, so that we can use DeepEqual.
+	xCopy := x
+	yCopy := y
```

Comment:
> Good call, the shallow copy also works here.  I pushed an update with the shallow copy but I can also use the conversion func method.  How does this look?
> 
```
	compare := conversion.EqualitiesOrDie(
		func(a, b metav1.OwnerReference) bool {
			return a.String() == b.String()
		},
		func(a, b metav1.ObjectMeta) bool {
			if a.Namespace != b.Namespace {
				return false
			}
			for k, v := range a.Labels {
				if b.Labels[k] != v {
					return false
				}
			}
			for k, v := range b.Labels {
				if a.Labels[k] != v {
					return false
				}
			}
			return true
		},
```

That LGTM -  I would go with the above to avoid even creating those shallow copies.

[Other than that it LGTM too.]

Reply:
+1

Reply:
Okay thanks!  Pushed